source/common/http/utility.h

```diff
@@ -626,6 +626,33 @@ getMergedPerFilterConfig(const Http::StreamFilterCallbacks* callbacks,
   return merged;
 }
 
+/**
+ * Return all the available per route filter configs.
+ *
+ * @param callbacks The stream filter callbacks to check for route configs.
+ *
+ * @return The all available per route config. The returned pointers are guaranteed to be non-null
+ * and their lifetime is the same as the matched route.
+ */
+template <class ConfigType>
+absl::InlinedVector<const ConfigType*, 3>
+getAllPerFilterConfig(const Http::StreamFilterCallbacks* callbacks) {
```

Comment:
First, Returning the vector will not introduce the copy. it will be RVO or at worst is move.
Second, Prefer Return Values to Output Parameters https://abseil.io/tips/176